src/utilities/xml/XMLValidator.cpp

```diff
@@ -0,0 +1,438 @@
+
+  auto schematron_filename_str = openstudio::toString(curDir / "iso_svrl_for_xslt1.xsl");
+  xsltStylesheet* style = xsltParseStylesheetFile(detail::xml_string(schematron_filename_str));
+
+  xmlDoc* res = xsltApplyStylesheet(style, curdoc, params);
+
+  xsltFreeStylesheet(style);
+  xmlFreeDoc(curdoc);
+
+  openstudio::filesystem::remove(curDir / "iso_svrl_for_xslt1.xsl");
+  openstudio::filesystem::remove(curDir / "iso_schematron_skeleton_for_xslt1.xsl");
+
+  return res;
+}
+
+openstudio::path schematronToXslt(const openstudio::path& schemaPath) {
+
+  // This replicates what happens when you do:
+  // from lxml.isoschematron import Schematron;
+  // s = Schematron(file='schematron.sct', store_xslt=True)
+  // with open('schematron.xslt', 'w') as f:
+  //     f.write(str(s._validator_xslt))
+
+  auto filename_str = openstudio::toString(schemaPath);
+  const auto* filename = filename_str.c_str();
+  xmlDoc* schematronXmlDoc = xmlParseFile(filename);
+
+  // For debugging
+  constexpr bool saveIntermediates = false;
+
+  auto saveXmlDocToFile = [](const openstudio::path& outputPath, xmlDoc* doc) {
+    auto save_filename_str = openstudio::toString(outputPath);
+    const auto* save_filename = save_filename_str.c_str();
+    const int result = xmlSaveFormatFileEnc(save_filename, doc, "UTF-8", 1);  // 1 means format=true
+    if (result == -1) {
+      // xmlSaveFormatFileEnc returns the number of bytes written or -1 if it failed
+      LOG_FREE_AND_THROW("schematronToXslt", "Writing to file failed for " << openstudio::toString(outputPath));
+    }
+  };
+
+  // Extract
+
+  const char* params[16 + 1];
+  int nbparams = 0;
+  params[nbparams] = nullptr;
+  xmlSubstituteEntitiesDefault(1);
+  xmlLoadExtDtdDefaultValue = 1;
+
+  // include
+  xmlDoc* withIncludes = applyEmbeddedXSLT(":/xml/resources/iso_dsdl_include.xsl", schematronXmlDoc, params);
+  if constexpr (saveIntermediates) {
+    saveXmlDocToFile(schemaPath.parent_path() / "1_withIncludes.xslt", withIncludes);
+  }
+
+  // expand
+  xmlDoc* withExpand = applyEmbeddedXSLT(":/xml/resources/iso_dsdl_include.xsl", withIncludes, params);
+  if constexpr (saveIntermediates) {
+    saveXmlDocToFile(schemaPath.parent_path() / "2_withExpand.xslt", withIncludes);
+  }
+
+  // compile: this one uses an xsl:import ... not sure how to merge them...
+  // xmlDoc* compiled = applyEmbeddedXSLT(":/xml/resources/iso_svrl_for_xslt1.xsl", withExpand, params);
+  xmlDoc* compiled = applyEmbeddedXSLTWithImports(schemaPath.parent_path(), withExpand, params);
+
+  openstudio::path xsltPath = schemaPath.parent_path() / openstudio::toPath(openstudio::toString(schemaPath.stem()) + "_stylesheet.xslt");
+  // xsltPath.replace_extension(".xslt");
+  saveXmlDocToFile(xsltPath, compiled);
+
+  xmlFreeDoc(compiled);
+
+  return xsltPath;
+}
+
+detail::XMLInitializer& XMLValidator::xmlInitializerInstance() {
+  static detail::XMLInitializer xmlInitializer;
+  return xmlInitializer;
+}
```

Comment:
A local static, on a static function of the class.

* It will initialize the libs on the first use (and not initialize if we have zero use of XMLValidator)
* It will be destroyed only at program exit.

cf https://godbolt.org/z/r69bc66qc for a short example how what happens.